src/user/models/update-user.dto.ts

```diff
@@ -0,0 +1,4 @@
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateUserDTO } from './create-user.dto';
+
+export class UpdateUserDTO extends PartialType(CreateUserDTO) {}
```

Comment:
- **why not just use typescript native Partial type?** => Because  ts native Partial type works with types/interfaces not classes. This expression `export class UpdateUserDTO extends Partial<CreateUserDTO> {}` gives me an error `'Partial' only refers to a type, but is being used as a value here`. IF I want to use ts native Partial type, I should convert it like this `export class UpdateShippingAddressDTO implements Partial<CreateShippingAddressDTO> {}` but I cannot convert dtos from classes to interfaces since interfaces are removed when ts code is transpilled into js and hence Nest cannot refer to them at runtime to do validation.

- I found myself need to define models(like `IAuthTokenPayload`) and dtos so I combined them into one folder. It might be a bad idea. I will avoid this in future projects.